app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,78 @@
+class TripsController < ApplicationController
+
+  def index
+    if params[:passenger_id]
+      @user = Passenger.find_by(id: params[:passenger_id])
+      @trips = @user.trips
+      
+    elsif params[:driver_id]
+      @user = Driver.find_by(id: params[:driver_id])
+      @trips = @user.trips
+    else
+      @trips = Trip.all
+    end
+  end
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+  end
+
+  def create
+    @passenger = Passenger.find_by(id: params[:passenger_id])
+    @driver = Driver.find_by(status: true)
+    @trip = @passenger.trips.new()
```

Comment:
All of the work done in this method could be moved to a model method. A class method on `Trip` would probably work nicely, something like `Trip.request_for(passenger_id)`.

Also, this action doesn't render a view, so you shouldn't need to use instance variables.